### PR TITLE
fixed matrix issue in correlation function + error handling

### PR DESCRIPTION
**Pull request by vrushaket**
Hi @josdejong,
This PR will fix issue #3025 please check.

### CODE REVIEW

File: src/function/statistics/corr.js [hunk 1]
```diff
@@ -40,13 +44,32 @@ export const createCorr = /* #__PURE__ */ factory(name, dependencies, ({ typed,
    * @private
    */
   function _corr (A, B) {
-    if (Array.isArray(A[0]) && Array.isArray(B[0])) {
-      const correlations = []
+    const correlations = []
```
Comment (josdejong) [inline src/function/statistics/corr.js:47]: The `_corr` function contains quite some duplicate code right now. I think changing it again to only handle `Array` and not `Matrix` will simplify it and remove this duplication. So, basically, take `_corr` as it was there and add the validations and handling of 2D there? Note that there is a util function `arraySize` that you can use if needed.

File: src/function/statistics/corr.js [hunk 2]
```diff
@@ -13,8 +15,10 @@ export const createCorr = /* #__PURE__ */ factory(name, dependencies, ({ typed,
    * Examples:
    *
    *     math.corr([1, 2, 3, 4, 5], [4, 5, 6, 7, 8])     // returns 1
-   *     math.corr([1, 2.2, 3, 4.8, 5], [4, 5.3, 6.6, 7, 8])     // returns 0.9569941688503644
-   *     math.corr(math.matrix([[1, 2.2, 3, 4.8, 5], [1, 2, 3, 4, 5]]), math.matrix([[4, 5.3, 6.6, 7, 8], [1, 2, 3, 4, 5]])) // returns DenseMatrix [0.9569941688503644, 1]
+   *     math.corr([1, 2.2, 3, 4.8, 5], [4, 5.3, 6.6, 7, 8])     //returns 0.9569941688503644
+   *     math.corr([[1, 2.2, 3, 4.8, 5], [4, 5.3, 6.6, 7, 8]],[[1, 2.2, 3, 4.8, 5], [4, 5.3, 6.6, 7, 8]])   // returns [1,1]
+   *     math.corr(math.matrix([2, 4, 6, 8]), math.matrix([1, 2, 3, 6]))    // returns 0.9561828874675149
```
Comment (josdejong) [inline src/function/statistics/corr.js:20]: Can you remove the examples with `math.matrix`? They are duplicates of the Array examples, so don't add much

**Review by josdejong — COMMENTED** [review]
Thanks for picking this up! I like all the extra validations and the unit tests for that, thanks. I made two inline comments, can you have a look at that?

File: src/function/statistics/corr.js (hunk 2 above)
Comment (vrushaket) [reply src/function/statistics/corr.js]: I have removed examples with math.matrix

File: src/function/statistics/corr.js (hunk 1 above)
Comment (vrushaket) [reply src/function/statistics/corr.js:45]: Actually the code if for handling some cases like 1D array, 2D array and matrices. Changing it to previous code, will lose support for matrix.

Comment (josdejong) [reply src/function/statistics/corr.js:45]: The behavior for Array and Matrix _should_ be exactly the same. And both Array and Matrix can hold 1D, 2D, or n-D content.

I think this already works in your implementation: I changed the implementation of the Matrix signature as follows, and then all unit tests pass, except for one (and in that case, the unit test wrongly expects an Array back but that should be a Matrix).

```js
    'Matrix, Matrix': function (A, B) {
      const res = _corr(A.toArray(), B.toArray())
      return Array.isArray(res) ? matrix(res) : res
    }
```

Does that make sense?

Comment (vrushaket) [reply src/function/statistics/corr.js:45]: yes absolutely.

**Comment by vrushaket** [comment]
@josdejong changes has been made. please have a look.

**Comment by josdejong** [comment]
Thanks this looks 👌

**Comment by josdejong** [comment]
The fix is published now in `v11.11.2`. Thanks again Vrushaket!

**Pull request merged**